engine/apps/api/views/slack_team_settings.py

```diff
@@ -11,11 +11,11 @@
 
 class SlackTeamSettingsAPIView(views.APIView):
     authentication_classes = (PluginAuthentication,)
-    permission_classes = (IsAuthenticated, MethodPermission)
+    permission_classes = (IsAuthenticated, RBACPermission)
 
-    method_permissions = {
-        IsAdmin: ("PUT",),
-        AnyRole: ("GET",),
+    rbac_permissions = {
+        "get": [RBACPermission.Permissions.CHATOPS_READ],
+        "put": [RBACPermission.Permissions.OTHER_SETTINGS_WRITE],
```

Comment:
tldr; I think the best way to solve this is to create a third chatops permissions, `CHATOPS_UPDATE_SETTINGS` that has a "fallback role" of `ADMIN`.

I had set it to `OTHER_SETTINGS_WRITE` because the fallback role for that was `ADMIN` whereas `CHATOPS_WRITE` had a fallback role of `EDITOR`. When it was set to `CHATOPS_WRITE` it would break some of the existing authorization tests in the other [PR](https://github.com/grafana/oncall/pull/732) (because the fallback role was too permissive, it would've allowed editors to access these endpoints whereas previously it was only admins).

Reply:
This should also address your comment for the PR in `oncall-private`.